### PR TITLE
Fix reading Delta Lake files with encoded characters on Azure

### PR DESCRIPTION
**Pull request by findinpath**
## Description

Previously there was a custom logic employed for building split paths on Azure, but this logic not actually needed.

The paths, which are relative to the table location, are already RFC 2396 URIs, so there is no additional encoding
needed before sending them to the filesystem client.

Builds on top of https://github.com/trinodb/trino/pull/17264
Thank you @vinay-kl  

For reference, this is how a Databricks table corresponding to the SQL statements below looks on Azure

```
CREATE TABLE findinpath_uri_test (part string, y long)
USING delta
PARTITIONED BY (part)
LOCATION 'abfss://.../findinpath_uri_test/';

INSERT INTO findinpath_uri_test VALUES ('a=equal', 1);
INSERT INTO findinpath_uri_test VALUES ('a:colon', 2);
INSERT INTO findinpath_uri_test VALUES ('a+plus', 3);
INSERT INTO findinpath_uri_test VALUES ('a space', 4);
INSERT INTO findinpath_uri_test VALUES ('a%percent', 5);
INSERT INTO findinpath_uri_test VALUES ('a/forwardslash', 6);
```

image

This listing is the same when creating the table via Databricks on s3

image

## Additional context and related issues

Fixes https://github.com/trinodb/trino/issues/21019

Reverts the fix done by https://github.com/trinodb/trino/pull/15381 

https://github.com/delta-io/delta/blob/master/PROTOCOL.md#add-file-and-remove-file

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
# Delta Lake
* Fix failure when the partition value contains forward slash characters. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by ebyhr** [comment]
/test-with-secrets sha=322394a89bf650f45d2faa4d1c1085bca7c4e59e

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java [hunk 1]
```diff
@@ -331,7 +331,7 @@ public void testPathUriDecoding()
         String tableName = "test_uri_table_" + randomNameSuffix();
         registerTableFromResources(tableName, "deltalake/uri", getQueryRunner());
 
-        assertQuery("SELECT * FROM " + tableName, "VALUES ('a=equal', 1), ('a:colon', 2), ('a+plus', 3), ('a space', 4), ('a%percent', 5)");
+        assertQuery("SELECT * FROM " + tableName, "VALUES ('a=equal', 1), ('a:colon', 2), ('a+plus', 3), ('a space', 4), ('a%percent', 5), ('a/forwardslash', 6)");
```
Comment (pajaks) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java:334]: Could you add test that also writes value with `/`?

Comment (findinpath) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java:334]: Thank you @pajaks  for the suggestion.

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java [hunk 2]
```diff
@@ -382,7 +382,8 @@ public static Location buildSplitPath(Location tableLocation, AddFileEntry addAc
         Optional<String> scheme = tableLocation.scheme();
         if (scheme.isPresent() && (scheme.get().equals("abfs") || scheme.get().equals("abfss"))) {
             // Replace '+' with '%2B' beforehand. Otherwise, the character becomes a space ' ' by URL decode.
-            path = URLDecoder.decode(path.replace("+", "%2B"), UTF_8);
+            // Replace '%2F' with '%252F' in order to maintain also after URL decoding the original '%2F'
+            path = URLDecoder.decode(path.replace("+", "%2B").replace("%2F", "%252F"), UTF_8);
```
Comment (findepi) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java:386]: why isn't this solved in the filesystem layer?

Comment (findinpath) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java]: It is resolved now - i've taken the contribution from https://github.com/trinodb/trino/pull/17264

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAdlsConnectorSmokeTest.java [hunk 3]
```diff
@@ -132,7 +132,11 @@ protected void registerTableFromResources(String table, String resourcePath, Que
                     .filter(resourceInfo -> resourceInfo.getResourceName().startsWith(resourcePath + "/"))
                     .collect(toImmutableList());
             for (ClassPath.ResourceInfo resourceInfo : resources) {
-                String fileName = resourceInfo.getResourceName().replaceFirst("^" + Pattern.quote(resourcePath), quoteReplacement(targetDirectory));
+                String fileName = resourceInfo.getResourceName()
+                        .replaceFirst("^" + Pattern.quote(resourcePath), quoteReplacement(targetDirectory))
+                        // Replace '%' (corresponding to '%' character url encoded) with '%25' in order
```
Comment (findinpath) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAdlsConnectorSmokeTest.java:137]: Relates potentially to the false-negative found initially in https://github.com/trinodb/trino/issues/15376

**Review by electrum — APPROVED** [review]

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
/test-with-secrets sha=83f8ea53de042a89104174ce59d8326b84bad4b3

**Review by findepi — APPROVED** [review]

**Pull request merged**